packages/core/src/baseCogniteClient.ts

```diff
@@ -58,14 +64,32 @@ export interface ApiKeyLoginOptions extends Project {
   apiKey: string;
 }
 
-export const AAD_OAUTH = 'AAD_OAUTH';
-export const CDF_OAUTH = 'CDF_OAUTH';
-export const ADFS_OAUTH = 'ADFS_OAUTH';
-export type AuthFlowType =
-  | typeof AAD_OAUTH
-  | typeof CDF_OAUTH
-  | typeof ADFS_OAUTH;
+export type AAD_OAUTH = {
+  type: 'AAD_OAUTH';
```

Comment:
I really like the discriminated union approach.
However, could we still have type strings as constants? `const AAD_OAUTH = 'AAD_OAUTH'`

Reply:
I will make the change in my PR for now which is mostly identical except for missing the tests in this one

Reply:
but why do you want it btw .. the union will take care of it never matching anything else

Reply:
like this 

```
export const AAD_OAUTH_TYPE: 'AAD_OAUTH' = 'AAD_OAUTH';
export type AAD_OAUTH = {
  type: typeof AAD_OAUTH_TYPE;
  options: OAuthLoginForAADOptions;
};
```

not sure if I see the value, is it for better vanilla JS interop?